src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution_ArgsToParameters.cs

```diff
@@ -164,6 +174,44 @@ public ImmutableArray<int> ToImmutableArray()
                 ArgumentAnalysisResult.NormalForm(argsToParameters.ToImmutableArray());
         }
 
+        private static int? CheckForBadNonTrailingNamedArgument(AnalyzedArguments arguments, ParameterMap argsToParameters, ImmutableArray<ParameterSymbol> parameters)
+        {
+            // Is there any named argument used out-of-position and followed by unnamed arguments?
+
+            // If the map is trivial then clearly not.
+            if (argsToParameters.IsTrivial)
+            {
+                return null;
+            }
+
+            // Find the first named argument which is used out-of-position or a params parameter
+            int foundPosition = -1;
+            int length = argsToParameters.Length;
+            for (int i = 0; i < length;  i++)
+            {
+                if (arguments.Name(i) != null && 
+                    (argsToParameters[i] != i || parameters[argsToParameters[i]].IsParams))
```

Comment:
>parameters[argsToParameters[i]].IsParams [](start = 49, length = 40)

This check is not complete, see all the conditions that parameter must meet in OverloadResolution.IsValidParams. Please add a test case for an affected scenario. Note that we shouldn't be even checking for params unless the candidate is used in expanded form. Please add a test for that as well.
It looks like OverloadResolution.GetParameterType should be adjusted as well.